File: src/pt/joaomneto/ffgbutil/adventure/impl/util/DiceRoll.java
package pt.joaomneto.ffgbutil.adventure.impl.util;

public class DiceRoll {

	final Integer d1;
	final Integer d2;

	public Integer getD1() {
		return d1;
	}

	public Integer getD2() {
		return d2;
	}

	public DiceRoll(Integer d1, Integer d2) {
		super();
		this.d1 = d1;
		this.d2 = d2;
	}

	public Integer getSum(){
		return d1+d2;
	}

	@Override
	public String toString() {
		return "DiceRoll{" +
				"d1=" + d1 +
				", d2=" + d2 +
				", total="+ (d1+d2) +"}";
	}
}

File: src/main/java/pt/joaomneto/titancompanion/adventure/impl/util/DiceRoll.kt
package pt.joaomneto.titancompanion.adventure.impl.util

class DiceRoll(val d1: Int, val d2: Int) {

    val sum: Int
        get() = d1 + d2

    val isDouble: Boolean
        get() = d1 == d2

    override fun toString(): String {
        return "DiceRoll{" +
            "d1=" + d1 +
            ", d2=" + d2 +
            ", total=" + (d1 + d2) + "}"
    }
}
